src/modal/modal.js

```diff
@@ -358,9 +360,26 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.transition'])
             var templateAndResolvePromise =
               $q.all([getTemplatePromise(modalOptions)].concat(getResolvePromises(modalOptions.resolve)));
```

Comment:
That's a good suggestion, but I don't think it can work because `$modalStack` isn't updated until `templateAndResolvePromise` is resolved.